### PR TITLE
AB-340: Import MusicBrainz data in a separate musicbrainz schema in AB database

### PR DESCRIPTION
**Pull request by rsh7**
* To initialize the musicbrainz database:
`./develop.sh run --rm  webserver python2 manage.py init_mb_db`

* To start the import:
`./develop.sh run --rm  webserver python2 manage.py import_musicbrainz_db`

### CODE REVIEW

**Comment by rsh7** [comment]
Remaining things to do:
1. Add docstrings explaining work of all the functions.
2. Add comments in the entire code and more clean up.

File: db/import_mb_data.py [hunk 1]
```diff
@@ -0,0 +1,1512 @@
...
+                ON artist_credit.id = recording.artist_credit
+                WHERE recording.gid in :gids OR artist_credit.id in :data
+    """)
+    MB_release_fk_artist_credit = []
+    for value in MB_release_data:
+        MB_release_fk_artist_credit.append(value[3])
+    MB_release_fk_artist_credit = list(set(MB_release_fk_artist_credit))
+
+    result = connection.execute(artist_credit_query, {'gids': tuple(gids_in_AB), 'data': tuple(MB_release_fk_artist_credit)})
+    global MB_artist_credit_data
```
Comment (paramsingh) [inline db/import_mb_data.py]: Please don't use global, return this data and then use it.

File: db/import_mb_data.py [hunk 2]
```diff
@@ -0,0 +1,1512 @@
...
+    global MB_release_status_data
+    MB_release_status_data = result.fetchall()
+
+
+def load_release_group_primary_type(connection, gids_in_AB):
+    release_group_primary_type_query = text("""
+        SELECT DISTINCT release_group_primary_type.id, release_group_primary_type.name,
+               release_group_primary_type.parent, release_group_primary_type.child_order,
+               release_group_primary_type.description, release_group_primary_type.gid
+        FROM release_group_primary_type INNER JOIN release_group
```
Comment (paramsingh) [inline db/import_mb_data.py]: this query should be indented.

File: db/import_mb_data.py (hunk 1 above)
Comment (paramsingh) [reply db/import_mb_data.py]: Should fix this in every `load` function.

File: db/import_mb_data.py [hunk 3]
```diff
@@ -0,0 +1,1512 @@
...
+    print("INSERTED end_area data\n")
+
+
+def write_artist(transaction, connection):
+    artist_query = text("""
+      INSERT INTO musicbrainz.artist
+          VALUES (:id, :gid, :name, :sort_name, :begin_date_year, :begin_date_month, :begin_date_day,
+                :end_date_year, :end_date_month, :end_date_day, :type, :area, :gender, :comment, :edits_pending,
+                :last_updated, :ended, :begin_area, :end_area)
+          ON conflict do nothing
```
Comment (paramsingh) [inline db/import_mb_data.py]: ON CONFLICT here should name the column which conflicts, and CONFLICT should be in caps.

File: db/import_mb_data.py [hunk 4]
```diff
@@ -0,0 +1,1512 @@
...
+    ]
+    connection.execute(recording_query, values)
+    transaction.commit()
+    print("INSERTED recording data\n")
+
+
+def write_recording_gid_redirect(transaction, connection):
+    recording_gid_redirect_query = text("""
+        INSERT INTO musicbrainz.recording_gid_redirect
+            VALUES (:gid, :new_id, :created)
```
Comment (paramsingh) [inline db/import_mb_data.py]: indentation.

File: db/import_mb_data.py [hunk 5]
```diff
@@ -0,0 +1,1512 @@
...
+
+
+
+
+# FUNCTION TO CALL ALL INSERTS
+
+def insert_MB_data_AB():
+    with db.engine.connect() as connection:
+        if MB_artist_credit_data:
+            transaction = connection.begin()
```
Comment (paramsingh) [inline db/import_mb_data.py]: Do we need a transaction for each insert?

File: manage.py [hunk 6]
```diff
@@ -179,6 +181,15 @@ def remove_admin(username):
         sys.exit(1)
 
 
+@cli.command()
+def init_mb_db():
```
Comment (paramsingh) [inline manage.py]: this doesn't look needed ?

File: db/import_mb_data.py [hunk 7]
```diff
@@ -0,0 +1,1512 @@
...
+    INNER JOIN recording
+            ON track.recording = recording.id
+         WHERE recording.gid in :gids
+    """)
+    result = connection.execute(track_query, {'gids': tuple(gids_in_AB)})
+    global MB_track_data
+    MB_track_data = result.fetchall()
+
+
+print("--------------------------------------------------------------------------------------------------")
```
Comment (paramsingh) [inline db/import_mb_data.py]: remove this.

File: db/import_mb_data.py [hunk 8]
```diff
@@ -0,0 +1,1512 @@
...
+        except ValueError:
+            print("No Data found for the recordings")
+
+    insert_MB_data_AB()
+    print("--------------------------------DONE-----------------------------------")
+
+
+def start_import():
+    with db.engine.begin() as connection:
+        lowlevel_query = text("""SELECT gid from lowlevel""")
```
Comment (paramsingh) [inline db/import_mb_data.py]: Could we get these gids in batches too?

``` 
SELECT gid  FROM lowlevel ORDER BY id OFFSET 0 LIMIT 10000
SELECT gid  FROM lowlevel ORDER BY id OFFSET 0 LIMIT 10000
```
and so on.

File: db/import_mb_data.py [hunk 9]
```diff
@@ -0,0 +1,1512 @@
...
+        gids = connection.execute(lowlevel_query)
+        gids = gids.fetchall()
+        gids_in_AB = [value[0] for value in gids]
+        no_of_rows = len(gids_in_AB)
+        start = 0
+        rows_to_fetch = 10000
```
Comment (paramsingh) [inline db/import_mb_data.py]: we should make this a configurable variable in config.py

File: db/import_mb_data.py (hunk 1 above)
Comment (rsh7) [reply db/import_mb_data.py]: Done.

File: db/import_mb_data.py (hunk 5 above)
Comment (rsh7) [reply db/import_mb_data.py]: Removed the integrity check part as we are using 'on conflict action'.

File: db/import_mb_data.py (hunk 8 above)
Comment (rsh7) [reply db/import_mb_data.py]: Added.

**Pull request merged**